newDB.sql

```diff
@@ -1,97 +1,110 @@
+  state varchar(255) not null , -- check constraint for state and city
+  city varchar(255) not null,
+  address varchar(255) not null,
+  postal_code int not null,
+  created_at timestamp without time zone not null default now(),
+  updated_at timestamp without time zone
+  check(updated_at > created_at)
 );
 
-create table Account_Master (
+create table Customers (
   id bigserial primary key,
-  Account_Type varchar(255) not null,
-  prefix varchar(255) not null,
-  Min_balance decimal(10,2) not null,
-  interest decimal(10,2) not null
+  branch_id bigint not null references Branches(id),
+  first_name varchar(255) not null,
+  last_name varchar(255) not null,
+  country text not null,
+  state text not null, -- check constraint for state and city
+  city text not null,
+  address text [] not null,
+  postal_code int not null,
+  created_at timestamp without time zone not null default now(),
+  updated_at timestamp without time zone,
+  check(updated_at > created_at)
 );
 
-create table Customer (
-  CustomerID int not null primary key,
-  IFS_Code varchar(255) not null references Branch(IFS_Code),
-  F_Name varchar(255) not null,
-  L_Name varchar(255) not null,
-  Login_ID varchar(255) not null,
-  Acc_Pass varchar(255) not null,
-  Trans_Pass varchar(255) not null,
-  Acc_Status varchar(255) not null,
-  City varchar(255) not null,
-  State varchar(255) not null,
-  Country varchar(255) not null,
-  Acc_Open_Date date not null,
-  Last_login date not null
-);
+create index branch_idx on Customers(branch_id);
 
 create table Accounts(
-  Account_No varchar(255) not null primary key,
-  customer_ID int not null references Customer(CustomerID),
-  acc_status varchar(255) not null,
-  acc_open_date date not null,
-  account_type varchar(255) not null,
-  account_balance decimal not null
+  id bigserial primary key,
+  number bigint not null unique,
+  atm_pin int not null default 1234,
+  customer_id bigint not null references Customers(id),
+  status varchar(255) not null,
+  type text not null,
+  balance float not null,
+  created_at timestamp without time zone not null default now(),
+  updated_at timestamp without time zone,
+  check(created_at < updated_at)
 );
 
-create table Employee (
-  Emp_ID int not null primary key,
-  Emp_Name varchar(255) not null,
-  Login_ID varchar(255) not null,
-  password varchar(25) not null,
-  Email_ID varchar(255) not null,
-  Contact_No varchar(25) not null,
-  Create_Date date not null,
-  Last_Login date not null
-);
+create index accounts_customer_id_index on Accounts(customer_id);
 
-create table Loan (
-  Loan_ID int not null primary key,
-  Loan_Type varchar(25) not null,
-  Loan_Amount varchar(25) not null,
-  Customer_ID int not null references Customer(CustomerID),
-  Interest float not null,
-  Start_Date date not null
+create table Loans (
+  id bigserial primary key,
+  customer_id bigint not null references Customers(id),
+  type text not null,
+  amount float not null,
```

Comment:
Is this right data type for storing money? How float and doubles work in any language?

Reply:
money data type should be used.